### PR TITLE
Add <ReferenceOneField queryOptions> support

### PR DESCRIPTION
**Pull request by fzaninotto**

### CODE REVIEW

**Review by WiXSL — APPROVED** [review]

File: packages/ra-core/src/controller/field/useReferenceOneFieldController.tsx [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+> {
     record?: RaRecord;
     reference: string;
     source?: string;
     target: string;
     sort?: SortPayload;
     filter?: any;
+    queryOptions?: UseQueryOptions<{
+        data: RecordType[];
+        total: number;
```
Comment (WiXSL) [inline packages/ra-core/src/controller/field/useReferenceOneFieldController.tsx:20]:
```suggestion
        total?: number;
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/field/ReferenceOneField.tsx [hunk 2]
```diff
@@ -98,6 +104,10 @@ export interface ReferenceOneFieldProps
     sort?: SortPayload;
     filter?: any;
     link?: LinkToType;
+    queryOptions?: UseQueryOptions<{
+        data: RecordType[];
+        total: number;
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/field/ReferenceOneField.tsx:109]:
```suggestion
        total?: number;
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-core/src/controller/field/useReferenceOneFieldController.tsx (hunk 1 above)
Comment (fzaninotto) [reply packages/ra-core/src/controller/field/useReferenceOneFieldController.tsx:20]: TypeScript disagrees

File: packages/ra-ui-materialui/src/field/ReferenceOneField.tsx (hunk 2 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/field/ReferenceOneField.tsx:109]: Same

**Review by slax57 — APPROVED** [review]

**Pull request merged**